### ISSUE TITLE
Cite Data: Various formats of citation are not accessable via URL

### ISSUE THREAD

**Issue by mheppler** [IQSS/dataverse#3140]
In the [Widgets project](https://github.com/IQSS/dataverse/pull/3118), changes to how a user can access the various formats of the dataset citation were implemented, with the intent to improve the UX. Additional improvements should be made to access these formats directly through a URL.

For example, the URL to access the BibTeX version is:

```
http://localhost:8080/dataset.xhtml?persistentId=doi:10.5072/FK2/ZLRYYX#
```

Instead of pointing to the dataset pg, ideally we'd have an URL that is access directly to the BibTex citation version, which could look something like:

```
http://localhost:8080/dataset.bib?persistentId=doi:10.5072/FK2/ZLRYYX
```

**Comment by pdurbin** [comment]
I don't believe there are any formal requirements for this feature but here are some quick notes from a conversation with @scolapasta 

- 4 formats: 
  - regular citation
  - bibtex
  - RIS
  - EndNote XML
- per dataset version, see `:latest-published` and friends at http://guides.dataverse.org/en/4.3.1/api/native-api.html#datasets
- drafts too but check with @posixeleni 
- should all open in browser with URL you can email to someone (if it's published, at least)
- able to download as a file
- see also https://scholar.google.com/scholar?hl=en&q=gary+king&btnG=&as_sdt=1%2C22&as_sdtp=
- make pull request against widgets branch

I said I'm spend a couple hours with the code, at least, and see how long I think it would take to implement what's desired. @posixeleni would do acceptance testing.

**Comment by posixeleni** [comment]
@pdurbin regarding your https://github.com/IQSS/dataverse/issues/3140#issuecomment-221683642 comment/question about drafts: yes on drafts being able to support export on citations since researchers may want to add their data citation in pre-publication (hence why we already have a DOI reserved).

**Comment by pdurbin** [comment]
Related: Include dataset citation in native API #2059

**Comment by pdurbin** [comment]
@scolapasta as promised I spent about two hours poking around in the code and I think this is way more than a 2-3 day effort. My guess is two weeks, maybe more. I went ahead and pushed my scratch work to c433395 and I hope that the todo's I left in that commit help explain some of the challenges. I recommend revisiting this issue once the work for #2608 is complete.

**Comment by rhijmans** [comment]
I would love to have this feature. That is, the option to programmatically download a RIS file given a URI. 

I use the API to download multiple files "bundle"s in the original format but I cannot add a citation file.  [It looks like](https://guides.dataverse.org/en/4.20/api/dataaccess.html) I can do that with the "all formats" bundle, but that is not a good option for me (confusion about what the original files are;  the need to download different file formats, sometimes file by file to deal with size limitations; additional disk space cost).

**Comment by pdurbin** [comment]
@rhijmans thanks for the thoughtful comment. As you may have heard, we are planning on [rewriting the front end of Dataverse](https://dataverse.org/blog/restructuring-dataverse-ui-single-page-application) such that we'll need APIs for any functionality that's not currently exposed as APIs (such as downloading citations files). It's early days but I'm sure we'll get to this eventually.

If you (or anyone reading this) is interested in making a pull request, please get in touch! As mentioned above, it looks like I pushed some scratch work a long time ago (c433395) that might help someone get started.

**Comment by pdurbin** [comment]
We need this for the new frontend:

- https://github.com/IQSS/dataverse-frontend/issues/63

**Comment by cmbz** [comment]
2024/09/09: Keeping

**Comment by pdurbin** [comment]
This feature was delivered in the following PR:

- #11163

Here's a preview of docs from https://preview.guides.gdcc.io/en/develop/api/native-api.html#get-citation-in-other-formats

![Image](https://github.com/user-attachments/assets/edcf1ebe-fdf9-489f-84fb-b2afbdaef066)

Closing.
